### PR TITLE
Initial support for the ESP32-C6-DevKitC-1

### PR DESCRIPTION
**Pull request by linkedupbits**
This PR adds support for the ESP32-C6-DevKitC-1. It has been tested uploading the binary outputs from esp-idf

### CODE REVIEW

File: src/esploader.ts [hunk 1]
```diff
@@ -1,1050 +1,1055 @@
...
-    } else if (this.chip.CHIP_NAME != "ESP8266") {
-      throw new ESPError("Soft resetting is currently only supported on ESP8266");
-    } else {
-      // running user code from stub loader requires some hacks
-      // in the stub loader
-      await this.command(this.ESP_RUN_USER_CODE, undefined, undefined, false);
-    }
-  }
-}
+import { ESPError } from "./error";
```
Comment (balloob) [inline src/esploader.ts:1]: You've changed the line ending on these file, causing each line to be marked as changed. Please revert that.

Comment (linkedupbits) [reply src/esploader.ts]: These should be reverted now.

**Comment by linkedupbits** [comment]
I have also tested this using a local esphome-web-tools dev container, after building a esptool-js-0.2.1.tgz and placing it locally.
The local esphome-web-tools is then able to successfully connect to the C6 via the local COM port and show trace logs.
(Uploading a base ESPHome image is not working yet)

**Comment by linkedupbits** [comment]
@balloob  - with the reverted line endings, is there anything outstanding on this PR?

**Comment by balloob** [comment]
I'm sorry, I just hang around this repo as we rely on it downstream for ESP Web Tools. I am not able to review/merge PRs.

**Comment by linkedupbits** [comment]
> I'm sorry, I just hang around this repo as we rely on it downstream for ESP Web Tools. I am not able to review/merge PRs.

Makes sense - the reason I am trying to get this merged is for eventual ESPHome support.

If you don’t mind, to incorporate into a dev ESPHome:
 - I have built this into a tgz, and included it into a local esp-web-tools, which runs and recognises the Esp32 correctly and uploads firmware 
 - I am having difficulty then building/running a local ESPHome referencing the local esp-web-tools, that uses the local esptools-he.

is there a guide for a setting up a “fully local “ dev stack, to make integrating new chipsets easier for new starters?

**Comment by linkedupbits** [comment]
@brianignacio5 are you able to review this PR?

**Comment by brianignacio5** [comment]
LGTM !

**Comment by linkedupbits** [comment]
> npm run lint
local output:
``` bash
npm run lint

> esptool-js@0.2.2 lint
> eslint . --ext .ts


/home/stephen/esp/linkedupbits/esptool-js/src/targets/esp32c3.ts
  70:34  warning  'loader' is defined but never used  @typescript-eslint/no-unused-vars
  74:33  warning  'loader' is defined but never used  @typescript-eslint/no-unused-vars

/home/stephen/esp/linkedupbits/esptool-js/src/targets/esp32c6.ts
  70:34  warning  'loader' is defined but never used  @typescript-eslint/no-unused-vars
  74:33  warning  'loader' is defined but never used  @typescript-eslint/no-unused-vars

/home/stephen/esp/linkedupbits/esptool-js/src/targets/esp32s2.ts
  78:33  warning  'loader' is defined but never used  @typescript-eslint/no-unused-vars

/home/stephen/esp/linkedupbits/esptool-js/src/targets/esp32s3.ts
  43:37  warning  'loader' is defined but never used  @typescript-eslint/no-unused-vars
  46:34  warning  'loader' is defined but never used  @typescript-eslint/no-unused-vars
  49:33  warning  'loader' is defined but never used  @typescript-eslint/no-unused-vars

✖ 8 problems (0 errors, 8 warnings)
```

Should be better now.

**Comment by brianignacio5** [comment]
LGTM !

**Pull request merged**